log4j-docgen/src/main/java/org/apache/logging/log4j/docgen/processor/DocGenProcessor.java

```diff
@@ -0,0 +1,692 @@
+import org.apache.logging.log4j.docgen.Description;
+import org.apache.logging.log4j.docgen.PluginAttribute;
+import org.apache.logging.log4j.docgen.PluginElement;
+import org.apache.logging.log4j.docgen.PluginSet;
+import org.apache.logging.log4j.docgen.PluginType;
+import org.apache.logging.log4j.docgen.ScalarType;
+import org.apache.logging.log4j.docgen.ScalarValue;
+import org.apache.logging.log4j.docgen.Type;
+import org.apache.logging.log4j.docgen.io.stax.PluginBundleStaxWriter;
+import org.apache.logging.log4j.plugins.Factory;
+import org.apache.logging.log4j.plugins.Namespace;
+import org.apache.logging.log4j.plugins.Plugin;
+import org.apache.logging.log4j.plugins.PluginBuilderAttribute;
+import org.apache.logging.log4j.plugins.PluginFactory;
+import org.apache.logging.log4j.plugins.validation.constraints.Required;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@ServiceProvider(value = Processor.class, resolution = Resolution.OPTIONAL)
+@SupportedAnnotationTypes("org.apache.logging.log4j.plugins.*")
+@SupportedSourceVersion(SourceVersion.RELEASE_17)
+@NullMarked
+public class DocGenProcessor extends AbstractProcessor {
+
+    private static final String MULTIPLICITY_UNBOUNDED = "*";
+    private static final CharSequence[] GETTER_SETTER_PREFIXES = {"get", "is", "set"};
+    private static final List<Class<? extends Annotation>> PROPERTY_ANNOTATION_TYPES = List.of(
+            PluginBuilderAttribute.class,
+            org.apache.logging.log4j.plugins.PluginAttribute.class,
+            org.apache.logging.log4j.plugins.PluginElement.class);
+    private static final List<Class<? extends Annotation>> FACTORY_ANNOTATION_TYPES =
+            List.of(Factory.class, PluginFactory.class);
+    /**
+     * Reference types from the {@code java.*} namespace that are described
+     * in {@code org/apache/logging/log4j/docgen/internal/configuration.xml}
+     */
+    private static final Set<String> KNOWN_SCALAR_TYPES = Set.of(
+            "java.lang.Boolean",
+            "java.lang.Character",
+            "java.lang.Byte",
+            "java.lang.Short",
+            "java.lang.Integer",
+            "java.lang.Long",
+            "java.lang.Float",
+            "java.lang.Double",
+            "java.lang.String");
+
+    // Plugins
+    private final Map<String, PluginType> pluginTypes = new TreeMap<>();
+    // Other interfaces
+    private final Map<String, TypeElement> abstractTypes = new TreeMap<>();
+    // Scalar types
+    private final Map<String, TypeElement> scalarTypes = new TreeMap<>();
+
+    private AsciidocConverter converter;
+    private DocTrees docTrees;
+    private Elements elements;
+    private Types types;
+    private Messager messager;
+    // Type corresponding to java.util.Collection
+    private DeclaredType collectionType;
+    // Type corresponding to java.lang.Enum
+    private DeclaredType enumType;
+
+    @SuppressWarnings({"DataFlowIssue", "unused"})
```

Comment:
`DataFlowIssue` is a redundant suppression, IDEA says.

Reply:
If I remove it, it starts complaining that I assign `null` to fields that are `@NonNull`.